internal/server/data/migrations.go

```diff
@@ -17,10 +17,8 @@ import (
 	"github.com/infrahq/infra/uid"
 )
 
-func migrate(db *gorm.DB) error {
-	opts := migrator.DefaultOptions
-	opts.InitSchema = preMigrate
-	m := migrator.New(db, opts, []*migrator.Migration{
+func migrations() []*migrator.Migration {
```

Comment:
I removed the `migrate` function and created this `migrations` function instead. This should allow a test to compare the list of migrations against the list of tests in the table, to find when we've missing a migration test.

It also found some potential bugs. We were using the `db` variable instead of the `tx` variable in some of these migrations.